tools/xmakescripts/rules/build_rules.lua

```diff
@@ -87,6 +104,19 @@ local CLANG_COMPILE_OPTIONS = {
     }
 }
 
+-- option to use libc++
+option("libcxx")
+    set_default(false)
+    set_showmenu(true)
+    set_description("Use libc++ instead of libstdc++")
+
+if has_config("libcxx") then
```

Comment:
sure, I'm just hacking this for testing different building method. Currently I'm prefer to do a all zig build for linux target because once we locked on a zig version, we will have a fixed clang, libc++ and glibc version for building from any distro or windows.

Reply:
there is also a hack on line 231 in this file. I'm not sure what is the best way to describe that.. but I just want xmake to treat zig cc as a clang-like tool and thus get all flags from clang, (and it would be nicer if I can set some flags for zig only).